lib/jxl/decode.cc

```diff
@@ -882,29 +889,28 @@ static size_t GetStride(const JxlDecoder* dec, const JxlPixelFormat& format,
   return stride;
 }
 
-static JxlDecoderStatus ConvertImageInternal(const JxlDecoder* dec,
-                                             const jxl::ImageBundle& frame,
-                                             const JxlPixelFormat& format,
-                                             void* out_image, size_t out_size,
-                                             JxlImageOutCallback out_callback,
-                                             void* out_opaque) {
+static JxlDecoderStatus ConvertImageInternal(
```

Comment:
Can we have documentation of what's this function doing? It does quite different things depending on `want_extra_channel`; if it is true we get the extra channel only; but if it is false we get the color image. Maybe worth it to  have a different function for the "want_extra_channel==true" case, since it uses a different set of parameters, but for example do we need the RGBBuffer if want_extra_channel is true? We should assert that num_channels==1 in that case too since that should have been handled before.

Reply:
Fixed for external_image with https://github.com/libjxl/libjxl/pull/325, and documented for this internal function. I believe the internal function is simple enough to not split it into 3 functions (RGB(A) case, extra channel case, and the shared implementation), and documentation is added.